kahypar/partition/bin_packing/bin_packing_utils.h

```diff
@@ -208,68 +245,5 @@ static inline void preassignFixedVertices(const Hypergraph& hg, const std::vecto
     }
   }
 }
-
-// Calculates the maximum weight required for a bin if rb_range_k bins are used for the hypernodes.
-static inline HypernodeWeight currentMaxBin(const Hypergraph& hypergraph, const PartitionID& rb_range_k) {
-  BinaryMinHeap<PartitionID, HypernodeWeight> queue(rb_range_k);
-  for (PartitionID j = 0; j < rb_range_k; ++j) {
-      queue.push(j, 0);
-    }
-
-  const std::vector<HypernodeID> hypernodes = nodesInDescendingWeightOrder(hypergraph);
-  for (const HypernodeID& hn : hypernodes) {
-    PartitionID bin = queue.top();
-    queue.increaseKeyBy(bin, hypergraph.nodeWeight(hn));
-  }
-
-  // the maximum is the heaviest bin, i.e. the last element in the queue
-  while (queue.size() > 1) {
-    queue.pop();
-  }
-  return queue.getKey(queue.top());
-}
-
-// Given an assignment of hypernodes to parts, calculates the maximum weight required for a bin
-// if separate bins are used for each part.
-static inline HypernodeWeight resultingMaxBin(const Hypergraph& hypergraph, const Context& context) {
-  ASSERT(!context.partition.use_individual_part_weights);
-
-  const PartitionID num_parts = context.initial_partitioning.k;
-
-  // initialize queues
-  std::vector<BinaryMinHeap<PartitionID, HypernodeWeight>> part_queues;
-  part_queues.reserve(num_parts);
-  for (PartitionID i = 0; i < num_parts; ++i) {
-    const PartitionID current_k = context.initial_partitioning.num_bins_per_part[i];
-    BinaryMinHeap<PartitionID, HypernodeWeight> queue(current_k);
-    for (PartitionID j = 0; j < current_k; ++j) {
-        queue.push(j, 0);
-    }
-    part_queues.push_back(std::move(queue));
-  }
-
-  // assign nodes
-  const std::vector<HypernodeID> hypernodes = bin_packing::nodesInDescendingWeightOrder(hypergraph);
-  for (const HypernodeID& hn : hypernodes) {
-    const PartitionID part_id = hypergraph.partID(hn);
-
-    ALWAYS_ASSERT(part_id >= 0 && part_id < num_parts,
-                  "Node not assigned or part id " << part_id << " invalid: " << hn);
-
-    const PartitionID bin = part_queues[part_id].top();
-    part_queues[part_id].increaseKeyBy(bin, hypergraph.nodeWeight(hn));
-  }
-
-  HypernodeWeight max = 0;
-  for(auto& queue : part_queues) {
-    // the maximum is the heaviest bin, i.e. the last element in the queue
-    while (queue.size() > 1) {
-      queue.pop();
-    }
-    max = std::max(queue.getKey(queue.top()), max);
-  }
-
-  return max;
-}
```

Comment:
This functionality is moved to bin_packer.h (currentBinImbalance/partitionIsDeeplyBalanced) in a slightly more general fashion: It is now applicable to individual part weights and it is abstracted over different load balancing algorithms (thus the need to call it with dynamic dispatch).